### PR TITLE
Fix SafeTarFile symlinks extraction

### PR DESCRIPTION
**Pull request by e3krisztian**
Based on the work of @AndrewFasano in PR #770 this should fix most of the tar symlink extraction problems reported.

Related issue: #769

### CODE REVIEW

File: unblob/handlers/archive/_safe_tarfile.py
Comment (qkaiser) [inline unblob/handlers/archive/_safe_tarfile.py:91]: We could add an assert here to make sure it is and be noisy if it's not, which is indicative of something wrong with `is_safe_path`.

```python
assert root == common_path
```

**Review by qkaiser — COMMENTED** [review]
Small comment otherwise LGTM. Nice work on tests.

File: unblob/handlers/archive/_safe_tarfile.py
Comment (e3krisztian) [reply unblob/handlers/archive/_safe_tarfile.py]: Yes, this made me thinking of adding an assertion as well.

However an assert would blow up extraction, instead it is explicitly reported now.

A fixup commit is added for review, it needs to be squashed before merge (if accepted).
Converting to draft to prevent accidental merge, but it is supposed to be complete and waiting for review.

**Comment by e3krisztian** [comment]
Rebased for linear history.

File: unblob/handlers/archive/_safe_tarfile.py
Comment (qkaiser) [inline unblob/handlers/archive/_safe_tarfile.py:86]: This would be easier to understand I think:

```python
if common_path != root:
    return None
```

But maybe I'm missing something.

**Review by qkaiser — APPROVED** [review]
As agreed, squash before merge.

File: unblob/handlers/archive/_safe_tarfile.py
Comment (e3krisztian) [reply unblob/handlers/archive/_safe_tarfile.py:86]: yeah, except `common_path` is no more with the last change.

**Pull request merged**
